hack/install-requirements.sh

```diff
@@ -31,6 +31,12 @@ if [[ ${platform} == "Linux" ]]; then
     curl -L -o /usr/local/bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
     chmod +x /usr/local/bin/jq
   fi
+
+  if ! which yq &>/dev/null; then
```

Comment:
Proposal:
Let's not install `yq` to `/usr/local/bin` but rather to a repo-local tools bin dir and use it for the make targets / hack scripts from there.
I find it quite annoying that `make install-requirements` overwrites your global tools (e.g. helm), especially if there are breaking changes between your global version and the version required by gardener, which is why I never use this target. It is also used in our pipelines, where it makes sense, but it's just annoying for developers.

We can use the same approach as controller-runtime, see [here](https://github.com/kubernetes-sigs/controller-runtime/blob/985e819db7af1cd17782ca7ad27644d0ca655054/Makefile#L38-L42) and respective [targets](https://github.com/kubernetes-sigs/controller-runtime/blob/985e819db7af1cd17782ca7ad27644d0ca655054/Makefile#L64-L77).

Had this on my list for a long time, but haven't been able to do it yet. But as you mentioned there are many breaking changes between yq 3 and 4, let's use this opportunity to at least install yq in a proper tools bin dir.
WDYT?